src/main/java/kr/codesqaud/cafe/controller/ArticleController.java

```diff
@@ -37,7 +45,35 @@ public String write(@ModelAttribute ArticleSaveRequest articleSaveRequest) {
 
     @GetMapping("/articles/{id}")
     public String getArticle(@PathVariable Long id, Model model) {
-        model.addAttribute("article", articleService.findById(id));
+        model.addAttribute("article", articleService.getArticleWithSurrounding(id));
         return "article/detail";
     }
+
+    @GetMapping("/articles/{id}/edit")
+    public String editArticle(@PathVariable Long id, Model model, HttpSession httpSession) {
```

Comment:
여기 argument에도 final keyword를 붙여주시면 어떨까요?
```suggestion
    public String editArticle(@PathVariable final Long id, final Model model, final HttpSession httpSession) {
```
참, 그러기 전에, final 키워드와 불변성은 어떤 관련이 있을까요? 불변성은 또 왜 중요할까요?

Reply:
final 키워드를 통해 한 번 생성된 객체의 내부 상태를 변경할 수 없도록 강제할 수 있기 때문에 불변성을 구현하는 데 사용하는 것으로 알고 있습니다
예상하지 못한 곳에서 값이 변경되는 것을 방지하여 예측 가능하고 안정적인 동작을 보장하기 때문에 잘 사용하면 좋다고 이해했는데, 아직 언제 어떻게 사용해야 할지 정확히 모르겠습니다 🥲 좀 더 공부해 볼게요!

Reply:
다음 영상을 찾아보세요.
https://m.youtube.com/watch?v=ej-bnXlHk-E